### ISSUE TITLE
Layout-dialect migration issues (thymeleaf 2->3,  layout-dialect 2.0-snapshot) 

### ISSUE THREAD

**Issue by robvanderboom** [ultraq/thymeleaf-layout-dialect#111]
Hi,
i am a developer and trying to migrate an existing fully working website from thymeleaf 2 to thymeleaf 3 including the layout-dialect.  After a bugfix in the thymeleaf-data-extras, everthing seems to work except the styling of the website. After some research it looks that the layout:decorator causes the issue that the content of the head-tag within the home.html (main template) page is always empty when adding a layout:decorator tag. Below the part of code causing the issue (this structure causes the issue on every page of the website).

When i remove the layout:decorator tag, the <head> part is included in the output result of the page as-is.
However when adding the layout:decorator basic template (see below), the output result of the page is always <head></head> (empty head-content).  Even when the layout/2col_wide_left template is empty, or whatever i  place in there the result output page shows an empty <head>.
Even if i also place a <head> <title>xxx</title></head> part in the base template (2col_wide_left), the <head></head> in the output result is empty.

Is verion 2.0-snapshot of layout-dialect not thymeleaf-3 ready ?  Or do i have to change something in my html templates (but then it should be explained in the migration guide ?).

Hope someone can help ;-)
```
!DOCTYPE html>
<html xmlns:th="http://www.thymeleaf.org"
      xmlns:layout="http://www.ultraq.net.nz/thymeleaf/layout"
      layout:decorator="'layout/2col_wide_left'">
    <head>
        <title>Title</title>
        [everything i put here like a href to load css is removed from the result,
          so also the title is not shown in the result of the page.]
    </head>
    <body>
.... etc.`
```

**Comment by retromuz** [comment]
Hi @robvanderboom, please have a look at issue [110](https://github.com/ultraq/thymeleaf-layout-dialect/issues/110).

**Comment by ultraq** [comment]
@robvanderboom, if you find that @retromuz's fix works, then there's a bug in my code that doesn't handle those self-closed elements properly, and I should fix that in a future snapshot version.

**Comment by robvanderboom** [comment]
Hi, no sorry the project already loses links and meta tags with />.  And i also do not get exceptions.
Wish that was the problem but unfortunately  not ;-)

**Comment by robvanderboom** [comment]
Loses=closes

**Comment by gochev** [comment]
is the snapshot version available on some maven repo ? dont see it on jitpack.io.

**Comment by zhanhb** [comment]
http://oss.sonatype.org/content/repositories/snapshots

**Comment by ultraq** [comment]
Hi @robvanderboom, sorry for taking so long to get back to you.  I've recently finished getting the latest 2.0 snapshot feature-complete and in the process got through more tests and use cases that the layout dialect is supposed to do, including the one you've raised the issue about re: losing elements in the `<head>` section of your template.

Have you tried the latest snapshot to see if it helps at all?

Aside from the filling-out of more features, I think the next biggest thing that has changed since posting of this issue and now is that I've added a bit of backwards-compatibility code so that you no longer have to wrap template names in single quotes to get the layout dialect to treat it as a string like I see in your example, `'layout/2col_wide_left`'.  In your case, the layout dialect will wrap something like `layout/2col_wide_left` as a Thymeleaf 3 fragment expression (wrapped in `~{...}`), and emit a warning in the logs that you might want to do that yourself to future-proof your code.

Aside from that, all I can see is that your example has a missing `<` for the DOCTYPE, but I think that's just a typo.
